### ISSUE TITLE
Forbidden action  - Call of protected function 'UseQuestLogSpecialItem()'

### ISSUE THREAD

**Issue by GhostReapersWorld** [N0ich/dialogkey-DF#45]
Hey there,

since the latest wow update I get this error when trying to use Quest Items from the Questlog. 

> 
> [ADDON_ACTION_FORBIDDEN] AddOn 'DialogKey' tried to call the protected function 'UseQuestLogSpecialItem()'.
> [string "@!BugGrabber/BugGrabber.lua"]:485: in function <!BugGrabber/BugGrabber.lua:485>
> [string "=[C]"]: in function 'UseQuestLogSpecialItem'
> [string "@Blizzard_ObjectiveTracker/Blizzard_ObjectiveTrackerShared.lua"]:95: in function <...ObjectiveTracker/Blizzard_ObjectiveTrackerShared.lua:88>
> 
> Locals:
> _ = Frame {
>  RegisterEvent = <function> defined @!BugGrabber/BugGrabber.lua:491
>  0 = <userdata>
>  UnregisterEvent = <function> defined @!BugGrabber/BugGrabber.lua:491
>  SetScript = <function> defined @!BugGrabber/BugGrabber.lua:491
> }
> event = "ADDON_ACTION_FORBIDDEN"
> events = <table> {
>  ADDON_ACTION_BLOCKED = <function> defined @!BugGrabber/BugGrabber.lua:557
>  ADDON_ACTION_FORBIDDEN = <function> defined @!BugGrabber/BugGrabber.lua:557
>  PLAYER_LOGIN = <function> defined @!BugGrabber/BugGrabber.lua:551
>  LUA_WARNING = <function> defined @!BugGrabber/BugGrabber.lua:566
>  ADDON_LOADED = <function> defined @!BugGrabber/BugGrabber.lua:511
> }

**Comment by Slackluster** [comment]
Yep, experiencing this too. I initially thought this issue would be resolved with 11.0.2 because the taunt issues were making a comeback, but it seems to be something else.

**Comment by kcoderhere** [comment]
```
Date: 2024-10-31 19:33:24
ID: 1
Error occured in: AddOn: DialogKey
Count: 1
Message: Error: AddOn DialogKey attempted to call a forbidden function (UseQuestLogSpecialItem()) from a tainted execution path.
Debug:
   [string "=[C]"]: UseQuestLogSpecialItem()
   [string "@Interface/AddOns/Blizzard_ObjectiveTracker/Blizzard_ObjectiveTrackerShared.lua"]:94:
      ...ObjectiveTracker/Blizzard_ObjectiveTrackerShared.lua:87
```

Looks like it's back :(

**Comment by FuriousProgrammer** [comment]
As far as I have been able to tell, this is a generic Tainted Execution Path error that DialogKey is being blamed for, but isn't actually *causing*. Nothing in the code touches the Quest Tracker itself, but there's probably something funky going on internally that links it with the Quest *Panel(s)* that the addon **does** touch.

Unfortunately, there's nothing I or anyone else can do to fix this, you just need to /reload or use the quest item from your bags directly.
